### DISCUSSION TITLE
[question] How to match whole words only?

### DISCUSSION THREAD

**Discussion by Mennaruuk** [searxng/searxng#977] [Q&A]
When searching with SearXNG, I often find results that are irrelevant because they include words in the query as part of other words or phrases. A hypothetical example: imagine results for Owl City’s _Fireflies_ showing up when searching for how to spark a fire. I have yet to find a parameter that makes SearXNG match for whole words only. Google has an equivalent feature called Verbatim, and [on this page](https://booleanstrings.com/2018/03/11/five-hidden-google-operators/) I found a Boolean string for Verbatim, but even after appending it to the SearXNG query URL, search results were not affected. I also tried enclosing the words in quotation marks, but SearXNG still returned similar results.

Is anyone aware of a workaround that allows me to search for the whole word, and not its being a part of other words or phrases?

**Comment by dalf** [comment]
https://booleanstrings.com/2018/03/11/five-hidden-google-operators/ suggest to use `&tbs=li:1` (page published in 2018).

It make a small difference (I'm not sure which result set is the best...):
* `https://www.google.com/search?hl=en&as_q=&as_epq=%22steven+jobs%22&as_oq=&as_eq=&as_nlo=&as_nhi=&lr=&cr=&as_qdr=all&as_sitesearch=&as_occt=any&safe=images&as_filetype=&tbs=li:1`
* `https://www.google.com/search?hl=en&as_q=&as_epq=%22steven+jobs%22&as_oq=&as_eq=&as_nlo=&as_nhi=&lr=&cr=&as_qdr=all&as_sitesearch=&as_occt=any&safe=images&as_filetype=&tbs=`

In *this* example quotes works : `!google "Steven jobs"` in a SearXNG instance returns the result about Steven Jobs.

I guess you have example where `tbs=li:1` provides better result?

**Reply by Mennaruuk** [reply]
Nope. For me, after scrolling, I eventually come across search results where random letters (not words) are bolded. I think at that point, SearXNG starts to find anything that might match my query rather than give me no results. Still, this only works with Google. Other search engines do not have the Verbatim tag, but they also don't suffer from this problem. I'm not sure why SearXNG does this. It isn't a very pressing problem, but when I do come across it, I'm forced to go to the individual search engines directly to search, which is not ideal. Thanks for your help @dalf!
